services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java

```diff
@@ -265,7 +263,9 @@ public boolean checkTokenValidForIntrospection(KeycloakSession session, RealmMod
             }
 
             if (valid) {
-                userSession.setLastSessionRefresh(Time.currentTime());
+                int currentTime = Time.currentTime();
+                userSession.setLastSessionRefresh(currentTime);
+                userSession.getAuthenticatedClientSessions().values().forEach(c -> c.setTimestamp(currentTime));
```

Comment:
This needs to be limited to `client`. Clients unrelated to this token should not be updated here.
Then the proposed behaviour would seem correct to me as it is consistent with refreshing tokens: https://github.com/keycloak/keycloak/blob/e94017f258dbdb8224704818d033c21fa7d44112/services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java#L332-L334
WDYT @mposolda?

Reply:
+1 to the fact that clientSession timestampt should be updated here and that it should be just timestamp of the clientSession, which is related to this particular client. It seems that the code referenced in this comment is outdated, but latest code in this PR in class TokenManager https://github.com/keycloak/keycloak/pull/7844/files#diff-fd8b1c47da5ec7ee42d5450d3f7b73cdeef9c08af26b825c9245e3e6052c5cdf  looks correct to me .

Reply:
Thank you for the confirmation Marek. Yes, I updated the piece of code to refresh just relevant clientSession.